packages/ngtools/webpack/src/transformers/import_factory.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+
+
+/**
+ * Given this original source code:
+ *
+ * import { NgModule } from '@angular/core';
+ * import { Routes, RouterModule } from '@angular/router';
+ *
+ * const routes: Routes = [{
+ *   path: 'lazy',
+ *   loadChildren: () => import('./lazy/lazy.module').then(m => m.LazyModule).
+ * }];
+ *
+ * @NgModule({
+ *   imports: [RouterModule.forRoot(routes)],
+ *   exports: [RouterModule]
+ * })
+ * export class AppRoutingModule { }
+ *
+ * NGC (View Engine) will process it into:
+ *
+ * import { Routes } from '@angular/router';
+ * const ɵ0 = () => import('./lazy/lazy.module').then(m => m.LazyModule);
+ * const routes: Routes = [{
+ *         path: 'lazy',
+ *         loadChildren: ɵ0
+ *     }];
+ * export class AppRoutingModule {
+ * }
+ * export { ɵ0 };
+ *
+ * The importFactory transformation will only see the AST after it is process by NGC.
+ * You can confirm this with the code below:
+ *
+ * const res = ts.createPrinter().printNode(ts.EmitHint.Unspecified, sourceFile, sourceFile);
+ * console.log(`### Original source: \n${sourceFile.text}\n###`);
+ * console.log(`### Current source: \n${currentText}\n###`);
+ *
+ * At this point it doesn't yet matter what the target (ES5/ES2015/etc) is, so the original
+ * constructs, like `class` and arrow functions, still remain.
+ *
+ */
+
+export function importFactory(
+  warningCb: (warning: string) => void,
+): ts.TransformerFactory<ts.SourceFile> {
+  return (context: ts.TransformationContext) => {
+    // Only transform files that contain `loadChildren`.
+    return (sourceFile: ts.SourceFile) => {
+      if (/\bloadChildren\b/.test(sourceFile.text)) {
+        const warning = `
+Found 'loadChildren' with a non-string syntax in ${sourceFile.fileName} but could not transform it.
+Make sure it matches the format below:
+
+loadChildren: () => import('IMPORT_STRING').then(m => m.EXPORT_NAME)
+
+Please note that only IMPORT_STRING and EXPORT_NAME can be replaced in this format.`;
+        const emitWarning = () => warningCb(warning);
+        const visitNode: ts.Visitor = (node: ts.Node) => {
+          // Only try to transform 'loadChildren' property assignments.
+          if (ts.isVariableDeclaration(node)) {
+            return replaceImport(node, context, emitWarning);
+          }
+
+          return ts.visitEachChild(node, visitNode, context);
```

Comment:
If ngc always creates the variable declaration at the root level then only the source file's children should need to be analyzed.  Walking the entire tree can be avoided.  The regex test above can be avoided as well since top level nodes are typically fairly minimal and testing a large block of text will be much less efficient.

Reply:
That is a better approach, yes. I modified the code to only recurse into `VariableStatement` to find `VariableDeclaration` within, since the structure is `VariableStatement -> VariableDeclarationList -> VariableDeclaration`. PTAL